src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC } from 'react';
+import { ModalProps, ModalSize } from './Modal.types';
+import { classNames } from '../../shared/utilities';
+import { BaseDialog } from '../Dialog/BaseDialog/BaseDialog';
+
+import styles from './modal.module.scss';
+
+export const Modal: FC<ModalProps> = ({
+    size = ModalSize.medium,
+    headerClassName,
+    bodyClassName,
+    actionsClassName,
+    modalClassName,
+    modalWrapperClassName,
+    ...rest
+}) => {
+    const modalClasses: string = classNames([
+        styles.modal,
+        modalClassName,
+        { [styles.small]: size === ModalSize.small },
+        { [styles.medium]: size === ModalSize.medium },
+        { [styles.large]: size === ModalSize.large },
```

Comment:
Need Fullscreen modal too, there are many scenarios and existing features in the product that require this.